be/src/formats/parquet/file_reader.cpp

```diff
@@ -438,6 +447,80 @@ StatusOr<bool> FileReader::_filter_group(const tparquet::RowGroup& row_group) {
             }
         }
     }
+    return false;
+}
+
+bool FileReader::_filter_group_with_more_filter(const tparquet::RowGroup& row_group) {
+    // runtime_in_filter, the sql-original in_filter and is_null/not_null filter will be in
+    // _scanner_ctx->conjunct_ctxs_by_slot
+    for (auto kv : _scanner_ctx->conjunct_ctxs_by_slot) {
+        StatisticsHelper::StatSupportedFilter filter_type;
+        for (auto ctx : kv.second) {
+            if (StatisticsHelper::can_be_used_for_statistics_filter(ctx, filter_type)) {
+                const TupleDescriptor& tuple_desc = *(_scanner_ctx->tuple_desc);
+                const std::vector<SlotDescriptor*>& slots = tuple_desc.slots();
+                SlotDescriptor* slot = nullptr;
+                for (auto s : slots) {
+                    if (s->id() == kv.first) {
+                        slot = s;
+                        break;
+                    }
+                }
+                std::unordered_map<std::string, size_t> column_name_2_pos_in_meta{};
```

Comment:
what will happend if slot==nullptr?

Reply:
that means we have a slot not in tuple, there must be some error, but we can add some defensive code.

Reply:
`slot->col_name(` will be crash here?

Reply:
Yes, it shouldn't be nullptr, but we'd better add some defensive code here, and will do it.